### PR TITLE
update note and filter

### PR DESCRIPTION
**Pull request by ShreyashMahajan**
In this PR, 
1. Update note is been added.
2. filter feature is added.

### CODE REVIEW

**Review by SaurabhChawda — APPROVED** [review]
Design and Code look good to me

File: src/services/filterServices.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const filterByDate = (noteDataList, type) => {
+    const newNoteList = [...noteDataList];
+    console.log(newNoteList)
```
Comment (dev-enforced) [inline src/services/filterServices.jsx:3]: Can remove this debugger

File: src/components/textEditor/textEditor.jsx [hunk 2]
```diff
@@ -28,6 +29,8 @@ export const TextEditor = () => {
 
     const submit = (e) => {
         e.preventDefault();
+        noteData = { ...noteData, dateAddition: new Date().toLocaleString() }
+        console.log('noteString', noteData);
```
Comment (dev-enforced) [inline src/components/textEditor/textEditor.jsx:33]: Similarly can remove it here too

File: src/services/filterServices.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    if (newNoteList.length !== 0) {
+        switch (type) {
+            case 'OLDEST_FIRST':
+                return newNoteList.sort((dateOne, dateSecond) => {
+                    const finalDate = (new Date(dateOne.dateAddition)).getTime() - (new Date(dateSecond.dateAddition)).getTime();
+                    console.log('fdsdsds one', (new Date(dateOne.dateAddition)).getTime());
+                    console.log('fdsdsds two', (new Date(dateSecond.dateAddition)).getTime());
```
Comment (dev-enforced) [inline src/services/filterServices.jsx:10]: Similarly can remove these too

**Review by dev-enforced — APPROVED** [review]
The filter functionalities have been implemented through a really understandable format of dispatches and filter utility functions. Good work @ShreyashMahajan . Do check out the live preview feature implementation later on as in mine it isn't working correctly

File: src/services/filterServices.jsx (hunk 1 above)
Comment (ShreyashMahajan) [reply src/services/filterServices.jsx:3]: ok thank you i will do that

File: src/services/filterServices.jsx (hunk 3 above)
Comment (ShreyashMahajan) [reply src/services/filterServices.jsx:10]: ok thank you

**Pull request merged**